--- Products/ZenModel/Device.py
+++ Products/ZenModel/Device.py
@@ -1344,6 +1344,9 @@
         """
         # Set production state on all components that inherit from this device
         ret = super(Device, self).setProdState(state, maintWindowChange, REQUEST)
+        # Patch from Jay Stanley to fix changes to productionState without needing daemon bounce
+        # See ticket ZEN-30167 - next line added
+        self._p_changed = True
         if REQUEST:
             audit('UI.Device.Edit', self, productionState=state,
                   maintenanceWindowChange=maintWindowChange)
